forwardproxy.go

```diff
@@ -306,16 +350,32 @@ func (fp *ForwardProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int,
 		if !fp.hideVia {
 			r.Header.Add("Via", strconv.Itoa(r.ProtoMajor)+"."+strconv.Itoa(r.ProtoMinor)+" caddy")
 		}
-		response, err := fp.httpTransport.RoundTrip(r)
-		if err != nil {
-			if response != nil {
-				if response.StatusCode != 0 {
-					return response.StatusCode, errors.New("failed to do RoundTrip(): " + err.Error())
+		if fp.outgoing.IPs != nil {
+			addrIP := SelectOutgoing(fp, r)
+			addr, _ := net.ResolveTCPAddr("tcp", addrIP+":0")
+			var transport = &http.Transport{DialContext: (&net.Dialer{LocalAddr: addr, Timeout: fp.dialTimeout}).DialContext}
```

Comment:
Ideal PR would include something like
```
if fp.explicitOutgoingIP {
    transport.LocalAddr = SelectOutgoing(r)
}
```
with like 15 additional lines of code in forwardproxy itself. (And 100 new LOC in tests)